src/legacy/ui/public/agg_types/number_list/number_list.tsx

```diff
@@ -0,0 +1,165 @@
+  const [ascendingError, setAscendingError] = useState(EMPTY_STRING);
+
+  // responsible for discarding changes
+  useEffect(
+    () => {
+      const updatedModels = getUpdatedModels(numberArray, models, numberRange);
+      if (validateAscendingOrder) {
+        const isOrderValid = validateOrder(updatedModels);
+        setAscendingError(
+          isOrderValid
+            ? i18n.translate('common.ui.aggTypes.numberList.invalidAscOrderErrorMessage', {
+                defaultMessage: 'The values should be in ascending order.',
+              })
+            : EMPTY_STRING
+        );
+      }
+      setModels(updatedModels);
+    },
+    [numberArray]
+  );
+
+  useEffect(
+    () => {
+      setValidity(!hasInvalidValues(models));
+    },
+    [models]
+  );
+
+  const onChangeValue = ({ id, value }: { id: string; value: string }) => {
+    const parsedValue = parse(value);
+    const { isValid, errors } = validateValue(parsedValue, numberRange);
+    setValidity(isValid);
+
+    const currentModel = models.find(model => model.id === id);
+    if (currentModel) {
+      currentModel.value = parsedValue;
+      currentModel.isInvalid = !isValid;
+      currentModel.errors = errors;
+    }
+
+    onUpdate(models);
+  };
+
+  // Add an item to the end of the list
+  const onAdd = () => {
+    const newArray = [...models, getNextModel(models, numberRange)];
+    onUpdate(newArray);
+  };
+
+  const onDelete = (id: string) => {
+    const newArray = models.filter(model => model.id !== id);
+    onUpdate(newArray);
+  };
+
+  const onBlur = (model: NumberRowModel) => {
+    if (model.value === EMPTY_STRING) {
+      model.isInvalid = true;
+    }
+    setTouched();
+  };
+
+  const onUpdate = (modelList: NumberRowModel[]) => {
+    setModels(modelList);
+    onChange(modelList.map(({ value }) => (value === EMPTY_STRING ? undefined : value)));
+  };
+
+  return (
+    <>
+      {models.map((model, arrayIndex) => (
+        <Fragment key={model.id}>
+          <NumberRow
+            isInvalid={showValidation ? model.isInvalid : false}
+            disableDelete={models.length === 1}
+            model={model}
+            labelledbyId={labelledbyId}
+            range={numberRange}
+            onDelete={onDelete}
+            onChange={onChangeValue}
+            onBlur={() => onBlur(model)}
+            autoFocus={models.length !== 1 && arrayIndex === models.length - 1}
+          />
+          {showValidation && model.isInvalid && model.errors && model.errors.length > 0 && (
+            <EuiFormErrorText>{model.errors.join('\n')}</EuiFormErrorText>
+          )}
+          <EuiSpacer size="s" />
+        </Fragment>
+      ))}
+      {showValidation && ascendingError && <EuiFormErrorText>{ascendingError}</EuiFormErrorText>}
+      <EuiSpacer size="s" />
+      <EuiFlexItem>
+        <EuiButton iconType="plusInCircle" fill={true} fullWidth={true} onClick={onAdd} size="s">
```

Comment:
This button is too prominent since it's not the primary action for the whole page. Change to:

```suggestion
        <EuiButtonEmpty iconType="plusInCircleFilled" fullWidth={true} onClick={onAdd} size="xs">
```

<img width="299" alt="Screen Shot 2019-05-20 at 14 36 15 PM" src="https://user-images.githubusercontent.com/549577/58044053-cf583800-7b0c-11e9-9a4f-38c09136668b.png">